mpl_interactions/controller.py

```diff
@@ -274,23 +274,32 @@ def save_animation(
         slider = self.controls[param]
         ipywidgets_slider = False
         if "Box" in str(slider.__class__):
-            ipywidgets_slider = True
             for obj in slider.children:
                 if "Slider" in str(obj.__class__):
                     slider = obj
-            N = int((slider.max - slider.min) / slider.step)
-            min_ = slider.min
-            max_ = slider.max
-            step = slider.step
-        elif isinstance(slider, mSlider):
+
+        if isinstance(slider, mSlider):
             min_ = slider.valmin
             max_ = slider.valmax
             if slider.valstep is None:
-                N = N_frames if N_frames else 200
-                step = (max_ - min_) / N
+                n_steps = N_frames if N_frames else 200
+                step = (max_ - min_) / n_steps
             else:
-                N = int((max_ - min_) / slider.valstep)
                 step = slider.valstep
+        elif "Slider" in str(slider.__class__):
+            ipywidgets_slider = True
+            min_ = slider.min
+            max_ = slider.max
+            step = slider.step
+        else:
+            raise NotImplementedError(
+                "Cannot save animation for slider of type %s".format(slider.__class__.__name__)
+            )
+
+        if step is None:
```

Comment:
why this check? I don't think we can get here without `step` being defined. Also if we do somehow get this far and `step` is undefined then the animation won't work becuase `step` is used inside of the function `f` below.

Reply:
Because `valstep` can be `None`
https://github.com/ianhi/mpl-interactions/blob/36e8446e7e4fe7903c2df0c1c864db92559b3619/mpl_interactions/controller.py#L288
Or, well, I'm not sure whether this is true, but there was this check originally:
https://github.com/ianhi/mpl-interactions/blob/5fa461359b1ab49a686cbc8d1c4e27a073454810/mpl_interactions/controller.py#L288-L293

Reply:
Oh hang on, forgot to remove that previous check

Reply:
So what I had in mind was this 7cd84d1. This PR is a becoming a bit of a mess, so perhaps best to squash-merge it.